src/layouts/sidebar-sidecar/utils/product-url-adjusters.ts

```diff
@@ -1,124 +1,183 @@
-import { Pluggable } from 'unified'
-import remarkPluginAdjustLinkUrls from 'lib/remark-plugin-adjust-link-urls'
 import isAbsoluteUrl from 'lib/is-absolute-url'
-import sentinelData from 'data/sentinel.json'
+import { ProductData } from 'types/products'
 
 /**
+ * Given some product data,
+ * Return a (url) => adjustedUrl function,
+ * intended for use in fixing up links in docs content.
+ *
+ * Note that these URL adjustments are expected to be necessary for
+ * every product during initial migration to HashiCorp Developer.
+ *
+ * If we want to remove the need for these URL adjustments, then
+ * at some point in the future, we could apply these as remark transforms
+ * using "remarkPluginAdjustLinkUrls" to the original source content
+ * (across ALL versions!), and ask authors to use the adjusted URL format
+ * going forward. Asana task:
+ * https://app.asana.com/0/1202097197789424/1202569127502507/f
+ */
+export function getProductUrlAdjuster(
+	productData: ProductData
+): (url: string) => string {
+	switch (productData.slug) {
+		/**
+		 * Sentinel is a bit of a special case,
+		 * since it did not have a dedicated dot-io domain
+		 */
+		case 'sentinel':
+			return (url: string) => rewriteSentinelDocsUrls(url, productData)
+		/**
+		 * Vault and Consul (and possibly others?)
+		 * have authored "/api" links which used to be redirected to "/api-docs",
+		 * but which we're now preferring to handle by rewriting here.
+		 */
+		case 'vault':
+		case 'consul':
+			return (url: string) => {
+				const withApiDocsFix = rewriteApiToApiDocs(url)
+				const withProductPrefix = rewriteDocsUrl(withApiDocsFix, productData)
+				return withProductPrefix
+			}
+		/**
+		 * All other products need their docs routes prefixed, as within docs
+		 * content, authors write links as if URLs are on the dot-io sites.
+		 * For example, in Waypoint content, authors write URLs like
+		 * "/docs/some-waypoint-page", which need to be adjusted to be
+		 * "/waypoint/docs/some-waypoint-page".
+		 */
+		default:
+			return (url: string) => rewriteDocsUrl(url, productData)
+	}
+}
+
+/**
+ * NOTE on rewriteDocsUrls:
+ * Authors write content as if it only exists for their product.
+ * For example, Waypoint content contains links that start with "/docs",
+ * but we need to be adjust these links to start with "/waypoint/docs".
+ *
+ * NOTE on /api to /api-docs:
  * We must fix /api URLs, which should really be /api-docs. This would ideally
  * be done in the source MDX. However, we need to support all past versions, so
  * even if we corrected the latest source, we'd likely still want to run this on
  * older content.
+ */
```

Comment:
Can these two comments be moved to the block comments above their associated functions? That will put them as close as possible to where their information is useful.

Reply:
✅ great call, moved these to the associated functions